dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -4526,6 +4546,264 @@ async fn cmd_db_validate_volume_references(
     Ok(())
 }
 
+async fn cmd_db_validate_regions(
+    datastore: &DataStore,
+    clean_up_orphaned_regions: bool,
+) -> Result<(), anyhow::Error> {
+    // *Lifetime note*:
+    //
+    // The lifetime of the region record in cockroachdb is longer than the time
+    // the Crucible agent's region is in a non-destroyed state: Nexus will
+    // perform the query to allocate regions (inserting them into the database)
+    // before it ensures those regions are created (i.e. making the POST request
+    // to the appropriate Crucible agent to create them), and it will request
+    // that the regions be deleted (then wait for that region to transition to
+    // the destroyed state) before hard-deleting the records in the database.
```

Comment:
Since we're in omdb anyway, can we verify this before doing anything destructive? (Would that amount to checking that there are no volumes that refer to the regions we're considering deleting?)

Reply:
I'm not sure there's a way to check - we won't delete regions that have an associated record in crdb, so we have no volume id to check.